pkgs/development/python-modules/discordpy/default.nix

```diff
@@ -27,13 +28,17 @@ buildPythonPackage rec {
   ] ++ lib.optionalString withVoice [
     libopus
     pynacl
+    ffmpeg
   ];
 
   patchPhase = ''
     substituteInPlace "discord/opus.py" \
       --replace "ctypes.util.find_library('opus')" "'${libopus}/lib/libopus.so.0'"
     substituteInPlace requirements.txt \
       --replace "aiohttp>=3.6.0,<3.8.0" "aiohttp>=3.6.0,<4"
+  '' + lib.optionalString withVoice ''
+    substituteInPlace "discord/player.py" \
+      --replace "executable='ffmpeg'" "executable='${ffmpeg}/bin/ffmpeg'"
```

Comment:
Creating a wrapper script would be more maintainable than patching the code, see [gb](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/tools/backup/gamerbackup/default.nix#L21) for an example.
Also a good opportunity to replace the find_library patch with LD_LIBRARY_PATH in the wrapper.